pkg/build/golang/Dockerfile.template

```diff
@@ -44,13 +44,18 @@ RUN cd ${PLAN_DIR} \
     && go env -w GOPROXY="${GO_PROXY}" \
     && CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o testplan ${TESTPLAN_EXEC_PKG}
 
+# Store module dependencies
+RUN cd ${PLAN_DIR} \
+  && go list -m all > ./testground_dep_list
```

Comment:
I'd prefer if we store it in the root (/), outside the plan's source code, to avoid unnecessary mixing.